### PR TITLE
Display React logo only on active tab

### PR DESCRIPTION
**Pull request by fivepointseven**
Associated Issue: #4773

### Summary of Changes

* Fixed displaying react logo on non-React tabs when React tab is selected

### CODE REVIEW

**Comment by fivepointseven** [comment]
This has room for improvement beyond just fixing the bug (I'd prefer to have framework logos on the appropriate tabs all the time like in a code editor) but I've been familiar with the project for about 30 minutes, so I only did the bug for now :D

**Comment by jasonLaster** [comment]
@fivepointseven I see what you're saying! And i see how you fixed the problem, which is a step in the right direction. I'd like to take a stab at showing the correct logo for all the tabs though. 

I'm traveling, so i'm a bit less responsive, but I bet you can try getting the source meta data for all the tabs. It's a bit inefficient the way we're doing it now, but we can refactor it later.

**Comment by fivepointseven** [comment]
@jasonLaster I tried doing that yesterday and came up with a bunch of different ways to do it actually, but the problem is that for some reason `isReactComponent` was `false` the whole evening for me even with the current code. Weird.
Anyway I’ll try to get that to work and commit it later today :)

**Comment by fivepointseven** [comment]
@jasonLaster okay, done :)
React logos are now consistently displayed:

File: src/components/Editor/Tabs.js [hunk 1]
```diff
@@ -484,16 +484,20 @@ class SourceTabs extends PureComponent<Props, State> {
 
 export default connect(
   state => {
-    const selectedSource = getSelectedSource(state);
-    const sourceId = selectedSource ? selectedSource.get("id") : "";
+    const sourceTabs = getSourcesForTabs(state);
+    const sourceTabsMetaData = {};
+    sourceTabs.forEach(source => {
+      const sourceId = source ? source.get("id") : "";
```
Comment (codehag) [inline src/components/Editor/Tabs.js]: im wondering, what is the case when a source does not have an id? do you have an example?

also, meta thought: it feels like the meta data belongs to the tab as a data structure too..

File: src/components/Editor/Tabs.js [hunk 2]
```diff
@@ -98,7 +97,7 @@ type Props = {
   startPanelCollapsed: boolean,
   endPanelCollapsed: boolean,
   searchOn: boolean,
-  sourceMetaData: SourceMetaDataMap
+  sourceTabsMetaData: any
```
Comment (codehag) [inline src/components/Editor/Tabs.js]: hmmm perhaps we can make a type annotation?

Comment (fivepointseven) [reply src/components/Editor/Tabs.js]: Roger that

File: src/components/Editor/Tabs.js (hunk 1 above)
Comment (fivepointseven) [reply src/components/Editor/Tabs.js]: This was copied from the latest code, so I'm not entirely sure:
```diff
- const selectedSource = getSelectedSource(state);
- const sourceId = selectedSource ? selectedSource.get("id") : "";
+ sourceTabs.forEach(source => {
+   const sourceId = source ? source.get("id") : "";
```
Is there a case where source doesn't have an id?

----
A funny thing about metadata btw, it's set on "LOAD_SOURCE_TEXT", which means that in order for `isReactComponent` to be set tab needs to be visited at least once (if you open a few tabs and reload the window, logos won't show up until you click on respective tabs). It doesn't affect UX that much, and when I tried moving it to reducer it disintegrated my CPU, so I probably didn't entirely understand how it works outside of Tabs component, so I didn't touch it.

Comment (jasonLaster) [reply src/components/Editor/Tabs.js]: @fivepointseven that's a good observation. We should probably re-think source metadata so it is a field in sources and tabs, so that when we re-show a tab on reload it is saved...

that way you don't need to revisit it.

Comment (fivepointseven) [reply src/components/Editor/Tabs.js]: @jasonLaster What's the starting point in source's "life"? Cause I tried adding that at `ADD_SOURCES` action but it ended up eating all of my resources over 10 seconds after loading so I probably did it wrong.

Comment (jasonLaster) [reply src/components/Editor/Tabs.js]: haha - yeah. we don't want to parse every source! it would kill us :)

don't worry about that bug now, I'll add some context in another issue

Comment (fivepointseven) [reply src/components/Editor/Tabs.js]: Okay, cool then. So, do you have any issues with this PR I should fix? :)

File: src/workers/parser/frameworks.js [hunk 3]
```diff
@@ -5,20 +5,22 @@
...
-    importObj =>
-      importObj.source === "react" &&
-      importObj.specifiers.some(specifier => specifier === "React")
+function importsReact(imports, identifiers) {
+  return (
+    imports.some(
+      importObj =>
+        importObj.source === "react" &&
+        importObj.specifiers.some(specifier => specifier === "React")
+    ) || identifiers.some(identifier => identifier.name === "React")
```
Comment (jasonLaster) [inline src/workers/parser/frameworks.js]: this is an interesting change. Why is this needed? Also, could we add some new tests that would fail w/o it

**Comment by jasonLaster** [comment]
Minor point, when you open your next PR use a non master branch as it's easier for us to pull down and to push changes to

https://github.com/devtools-html/debugger.html/blob/master/docs/pull-requests.md#git-workflow

File: src/workers/parser/frameworks.js (hunk 3 above)
Comment (jasonLaster) [reply src/workers/parser/frameworks.js]: I bet this change could be in its own PR actually.

**Comment by jasonLaster** [comment]
Just checked and it works nicely! 

Happy to land it after we discuss the parser change

File: src/workers/parser/frameworks.js (hunk 3 above)
Comment (fivepointseven) [reply src/workers/parser/frameworks.js]: This is what caused the problem I mentioned earlier:
>  for some reason isReactComponent was false the whole evening for me even with the current code

`import React` was expected here and it didn't trigger by `const React = require('react')`, so I added the identifier check as well

Comment (jasonLaster) [reply src/workers/parser/frameworks.js]: Ahh brilliant. Lets do this in a separate PR. It fixes another problem :)

Comment (jasonLaster) [reply src/workers/parser/frameworks.js]: Actually, #4737 is already working on it

Comment (fivepointseven) [reply src/workers/parser/frameworks.js]: Oh, didnt realize that
I just kinda did it cause it was interfering with what I was doing :D

Comment (jasonLaster) [reply src/workers/parser/frameworks.js]: Yep! I do that all the time too :)

glad you got to see the parser code. It's my favorite part of the codebase
